ribbon/ribbon/swap.py

```diff
@@ -165,7 +165,9 @@ def create_offer(self, offer: Offer, wallet: Wallet) -> str:
         offer.oToken = get_address(offer.oToken)
         offer.biddingToken = get_address(offer.biddingToken)
 
-        nonce = self.w3.eth.get_transaction_count(wallet.public_key)
+        nonce = self.w3.eth.get_transaction_count(
+            Web3.to_checksum_address(cast(str, wallet.public_key))
```

Comment:
Argument 1 to "get_transaction_count" of "Eth" has incompatible type "Optional[str]"; expected "Union[Address, ChecksumAddress, ENS]"

Reply:
Also added a cast to str because `to_checksum_address` does not expect None
(and by reviewing how Wallet is initialized, I don't think public_key can really be None)